cumulusci/tasks/salesforce/tests/test_ProfileGrantAllAccess.py

```diff
@@ -0,0 +1,366 @@
+        <apexClass>TestClass</apexClass>
+        <enabled>true</enabled>
+    </classAccesses>
+    <fieldPermissions>
+        <field>Account.TestField__c</field>
+        <editable>true</editable>
+        <readable>true</readable>
+    </fieldPermissions>
+    <pageAccesses>
+        <apexPage>TestPage</apexPage>
+        <enabled>true</enabled>
+    </pageAccesses>
+    <recordTypeVisibilities>
+        <recordType>Account.Business_Account</recordType>
+        <default>false</default>
+        <personAccountDefault>false</personAccountDefault>
+        <visible>true</visible>
+    </recordTypeVisibilities>
+    <recordTypeVisibilities>
+        <recordType>Account.HH_Account</recordType>
+        <default>true</default>
+        <personAccountDefault>true</personAccountDefault>
+        <visible>true</visible>
+    </recordTypeVisibilities>
+    <tabVisibilities>
+        <tab>NPSP_Settings</tab>
+        <visibility>DefaultOn</visibility>
+    </tabVisibilities>
+</Profile>
+"""
+
+PACKAGE_XML_BEFORE = """<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+    <types>
+        <members>*</members>
+        <members>Account</members>
+        <members>Campaign</members>
+        <members>CampaignMember</members>
+        <members>Contact</members>
+        <members>Lead</members>
+        <members>Opportunity</members>
+        <name>CustomObject</name>
+    </types>
+    <types>
+        <name>Profile</name>
+    </types>
+    <version>39.0</version>
+</Package>"""
+
+
+def test_run_task():
+    with tempfile.TemporaryDirectory() as tempdir:
+        with open(
+            os.path.join(CUMULUSCI_PATH, "cumulusci", "files", "admin_profile.xml"), "w"
+        ) as f:
+            f.write(PACKAGE_XML_BEFORE)
+
+        retrieve_dir = pathlib.Path(tempdir, "retrieve")
+        retrieve_dir.mkdir()
+        target_dir = retrieve_dir / "profiles"
+        target_dir.mkdir()
+
+        with open(target_dir / "Admin.profile", "wb") as f:
+            f.write(ADMIN_PROFILE_BEFORE)
+
+        task = create_task(
+            ProfileGrantAllAccess,
+            {
+                "record_types": [
+                    {
+                        "record_type": "Account.HH_Account",
+                        "default": True,
+                        "person_account_default": True,
+                    }
+                ],
+                "namespaced_org": True,
+                "include_packaged_objects": False,
+            },
+        )
+
+        task._retrieve = mock.Mock()
+        task._deploy = mock.Mock()
+        task._create_directories = mock.Mock()
+        task.retrieve_dir = retrieve_dir
+        task.deploy_dir = pathlib.Path(tempdir, "deploy")
+        task.deploy_dir.mkdir()
+
+        task()
+
+        dest_path = task.deploy_dir / "profiles" / "Admin.profile"
+        assert dest_path.exists()
+        assert dest_path.read_bytes().decode("utf-8") == ADMIN_PROFILE_EXPECTED.decode(
+            "utf-8"
+        )
```

Comment:
Why not just compare the bytes, or convert ADMIN_PROFILE_EXPECTED to text and use dest_path.read_text()?

Reply:
Tied myself in a knot with an earlier version of this unit test. Corrected to just compare the bytes.